### PR TITLE
Add proof mode to make file

### PR DESCRIPTION
**Pull request by mmsc2**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -92,6 +123,15 @@ compare_trace: $(CAIRO_RS_TRACE) $(CAIRO_TRACE)
...
 
+compare_trace_memory_proof: $(COMPILED_PROOF_TESTS) $(CAIRO_RS_TRACE_PROOF) $(CAIRO_TRACE_PROOF) $(CAIRO_RS_MEM_PROOF) $(CAIRO_MEM_PROOF)
+	cd tests; ./compare_vm_state_with_proof_mode.sh trace memory
+
+compare_trace_proof: $(CAIRO_RS_TRACE_PROOF) $(CAIRO_TRACE_PROOF)
+	cd tests; ./compare_vm_state_with_proof_mode.sh trace
+
+compare_memory_proof: $(CAIRO_RS_MEM_PROOF) $(CAIRO_MEM_PROOF)
+	cd tests; ./compare_vm_state_with_proof_mode.sh memory
+
```
Comment (fmoletta) [inline Makefile:134]: Please add these new targets to .PHONY

**Review by fmoletta — APPROVED** [review]
LGTM!

File: .gitignore [hunk 2]
```diff
@@ -5,5 +5,7 @@
 !cairo_programs/manually_compiled/*.json
 **/*.trace
 **/*.memory
+**/*.trace.proof
+**/*.memory.proof
```
Comment (Oppen) [inline .gitignore:9]: Just call them *.proof.{trace,memory} and use the same rule.

File: src/vm/runners/cairo_runner.rs [hunk 3]
```diff
@@ -711,19 +711,22 @@ impl CairoRunner {
...
+                            Relocatable::from((index as isize, seg_offset)),
+                            relocation_table,
+                        )?;
+                        let value = relocate_value(elem.clone(), relocation_table)?;
+                        if self.relocated_memory.len() <= relocated_addr {
+                            self.relocated_memory.resize(relocated_addr + 1, None);
+                        }
+                        self.relocated_memory[relocated_addr] = Some(value);
+                    }
                     None => self.relocated_memory.push(None),
```
Comment (Oppen) [inline src/vm/runners/cairo_runner.rs:730]: Should this still push?

File: tests/compare_vm_state_with_proof_mode.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    fi
+done
+
+if test $failed_tests = 0; then
+    echo "All $passed_tests tests passed; no discrepancies found"
+else
+    echo "Comparisons: $failed_tests failed, $passed_tests passed, $((failed_tests + passed_tests)) total"
+fi
+
+exit "${exit_code}"
```
Comment (Oppen) [inline tests/compare_vm_state_with_proof_mode.sh:51]: Why a separate script?

**Review by Oppen — CHANGES_REQUESTED** [review]
See comments

File: .gitignore [hunk 5]
```diff
@@ -5,7 +5,6 @@
 !cairo_programs/manually_compiled/*.json
 **/*.trace
 **/*.memory
-**/*.trace.proof
-**/*.memory.proof
+**/*.proof.{trace, memory}
```
Comment (Oppen) [inline .gitignore:8]:
```suggestion
```

**Review by Oppen — APPROVED** [review]

**Pull request merged**